test/API/H5_api_test_util.c

```diff
@@ -645,76 +642,71 @@ generate_random_dataspace(int rank, const hsize_t *max_dims, hsize_t *dims_out,
+ * be inserted as well. The caller is responsible
+ * for freeing the returned filename pointer with
+ * free().
+ *
+ * This function is only intended to be used from
+ * within a test function, as it uses the test
+ * function parameters to determine if tests are
+ * being run in a multi-threaded manner.
  */
 herr_t
-prefix_filename(const char *prefix, const char *filename, char **filename_out)
+prefix_test_filename(TestParams_t *test_params, const char *prefix, const char *filename,
+                     char **filename_out)
 {
     char  *out_buf       = NULL;
-    herr_t ret_value     = SUCCEED;
     int    chars_written = 0;
-#ifdef H5_HAVE_MULTITHREAD
-    thread_info_t *tinfo = NULL;
-#endif
+    herr_t ret_value     = SUCCEED;
 
+    if (!test_params) {
+        TestErrPrintf("invalid test parameters pointer\n");
+        ret_value = FAIL;
+        goto done;
+    }
     if (!prefix) {
-        printf("    invalid file prefix\n");
+        TestErrPrintf("invalid file prefix\n");
         ret_value = FAIL;
         goto done;
     }
     if (!filename || (*filename == '\0')) {
-        printf("    invalid filename\n");
+        TestErrPrintf("invalid filename\n");
         ret_value = FAIL;
         goto done;
     }
     if (!filename_out) {
-        printf("    invalid filename_out buffer\n");
+        TestErrPrintf("invalid filename_out buffer\n");
         ret_value = FAIL;
         goto done;
     }
 
-    if (TEST_EXECUTION_THREADED) {
-#ifdef H5_HAVE_MULTITHREAD
-
-        if ((tinfo = (thread_info_t *)pthread_getspecific(test_thread_info_key_g)) == NULL) {
-            printf("    failed to retrieve thread-specific info\n");
-            ret_value = FAIL;
-            goto done;
-        }
-
-        if ((out_buf = generate_threadlocal_filename(prefix, tinfo->thread_idx, filename)) == NULL) {
-            printf("    failed to generate thread-specific filename\n");
-            ret_value = FAIL;
-            goto done;
-        }
-
-#else
-        printf("    thread-specific filename requested, but multithread support not enabled\n");
+    if (NULL == (out_buf = malloc(H5_API_TEST_FILENAME_MAX_LENGTH))) {
+        TestErrPrintf("couldn't allocate filename buffer\n");
         ret_value = FAIL;
         goto done;
-#endif
-    } else {
-        if (NULL == (out_buf = malloc(H5_TEST_FILENAME_MAX_LENGTH))) {
-            printf("    couldn't allocated filename buffer\n");
-            ret_value = FAIL;
-            goto done;
-        }
+    }
 
-        if ((chars_written = HDsnprintf(out_buf, H5_TEST_FILENAME_MAX_LENGTH, "%s%s", prefix, filename)) <
-            0) {
-            printf("    couldn't prefix filename\n");
-            ret_value = FAIL;
-            goto done;
-        }
+    if (test_params->IsMtTest) {
+        /* Generate thread-local filename */
+        chars_written = HDsnprintf(out_buf, H5_API_TEST_FILENAME_MAX_LENGTH,
+                                   "%s%d%s", prefix, test_params->MtTestParams.ThreadID,
```

Comment:
Get thread ID from the `TestParams_t` structure rather than needing thread-local data